chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -480,129 +480,26 @@ Array ConcatenateImpl(const std::vector<Array>& arrays, int8_t axis) {
 Array Concatenate(const std::vector<Array>& arrays) { return ConcatenateImpl(arrays, 0); }
 
 Array Concatenate(const std::vector<Array>& arrays, nonstd::optional<int8_t> axis) {
-    if (axis.has_value()) {
-        return ConcatenateImpl(arrays, *axis);
-    }
-    std::vector<Array> raveled_arrays;
-    raveled_arrays.reserve(arrays.size());
-    std::transform(arrays.begin(), arrays.end(), std::back_inserter(raveled_arrays), [](const Array& array) {
-        Shape shape{array.GetTotalSize()};
-        return array.Reshape(shape);
-    });
-    return ConcatenateImpl(raveled_arrays, 0);
-}
-
-namespace {
-std::vector<Array> StackGrad(const Array& gout, int8_t axis) {
-    Shape shape{gout.shape()};
-    Strides strides{gout.strides()};
-    size_t dim = shape[axis];
-    int64_t step = strides[axis];
-    int64_t offset = gout.offset();
-    bool is_empty = gout.GetTotalSize() == 0;
-    shape.erase(shape.begin() + axis);
-    strides.erase(strides.begin() + axis);
-
-    std::vector<Array> gxs;
-    std::vector<ConstArrayRef> gxs_refs{};
-    gxs.reserve(dim);
-    gxs_refs.reserve(dim);
-    {
-        NoBackpropModeScope scope{};
-        Dtype dtype = gout.dtype();
-        Device& device = gout.device();
-        for (size_t i = 0; i < dim; ++i) {
-            gxs.emplace_back(internal::MakeArray(shape, strides, dtype, device, gout.data(), offset));
-            gxs_refs.emplace_back(gxs.back());
-            if (!is_empty) {
-                offset += step;
-            }
-        }
-    }
-
-    {
-        BackwardBuilder bb{"stack-grad", gout, gxs_refs};
-        if (BackwardBuilder::Target bt = bb.CreateTarget()) {
-            bt.Define([axis](BackwardContext& bctx) {
-                std::vector<Array> ggxs;
-                ggxs.reserve(bctx.output_count());
-                for (size_t i = 0; i < bctx.output_count(); ++i) {
-                    // TODO(imanishi): Check if bctx.output_grad(i) is not nullopt.
-                    ggxs.emplace_back(*bctx.output_grad(i));
-                }
-                bctx.input_grad() = Stack(ggxs, axis);
-            });
-        }
-        bb.Finalize();
+    if (!axis.has_value()) {
```

Comment:
Just reversed the condition (logic inside the indented scope) since the case where no axis is given is the "special" case.